src/pgn.js

```diff
@@ -315,18 +323,28 @@ function gameGetterImpl(impl, gameIndex) {
  */
 exports.pgnRead = function(pgnString, gameIndex) {
 	var stream = new TokenStream(pgnString, 0);
+	var errors = [];
 
 	// Parse all games (and return a Database object)...
 	if(arguments.length === 1) {
 		var games = [];
 		while(true) {
 			var currentPos = stream.currentPosition();
-			if(!doSkipGame(stream)) {
-				break;
+			try {
+				if(!doSkipGame(stream)) {
```

Comment:
The idea of `doSkipGame` is already to ignore all the tokens until the next "end-of-game" token, to avoid parsing the games, and to skip as many potential error as possible. Maybe the behavior implemented in `doSkipGame()` is still too strict, and in that case it might be adapted. But it is probably not a good idea to introduce a second method (`stream.skipGame()`) that do the same. I suggest to keep one of these, and to remove the other.

Reply:
Ok, will revisit. I saw some games in my database were completely stopping parsing and so tried to hack it first before I started to really add other things like pgnWrite and json support when I realized I really like this library!

Reply:
Looked at this some more - the issue is that doSkipGame() is calling consumeToken() which parses things like the comments and if there is a problem, throws an exception which stops all processing. I need a simple non-token parsing skip that just jumps to the next END_OF_GAME token, which is what the stream.skipGame() does. I think doSkipGame() is making sure that the game is valid, and to do that it needs to actually parse the tokens and values, so maybe stream.skipToEndOfGame() would be a better name for the "just ignore everything due to error until we get to the end of game token" and I continue to use both.

Reply:
I have replaced doSkipGame() with checkValidGame() and skipToNextGame() - doSkipGame() seemed to be doing both jobs and making the two things explicit made parsing easier.